### DISCUSSION TITLE
Hubs Enterprise AWS Stack Failed to Create

### DISCUSSION THREAD

**Discussion by biscuitsanddavey** [mozilla/hubs#4759] [Hubs Cloud Questions] [answered]
Have a consistent issue with an error while launching a new Hubs enterprise stack. My attempts are as follows:

- Attempt 1 - Launched stack but was rolled back because of this error. This was the first error in the events tab.
- Attempt 2 - Cleared CNAMES and tried again, same result.
- Attempt 3 - Cleared CNAMES and set to ignore errors to avoid rollback, same result without the rollback and Hubs did not successfully launch.

This is the first time I've had this issue launching a new stack on AWS and I cannot find any previous mentions of this error in these discussions, the Hubs Discord, or AWS Troubleshooting. Greatly appreciate any guidance or a resource to help me resolve the issue! 

<img width="997" alt="Screen Shot 2021-10-20 at 8 38 12 PM" src="https://user-images.githubusercontent.com/12155129/138191245-f885753b-7d53-47ae-9f02-9947f923e44a.png">

**Comment by mikemorran** [comment]
Not an answer, but I have also experienced this during three attempts to launch a Hubs personal stack today.

**Reply by pattersonbl2** [reply]
This would be related to AWS ending support for python2.7 for creating new lambda's. I am currently researching to see about a possible fix for this issue.

**Comment by wmurphyrd** [comment]
I believe the cause is here - there is one Lambda in the template with a Python 2 dependency https://github.com/mozilla/hubs-ops/blob/e5923ab37888ffdbc153ac02de97cb66e1bc0ee0/cloudformation/stack.yaml#L1402-L1450

AWS [announced EOL for Python 2](https://aws.amazon.com/blogs/compute/announcing-end-of-support-for-python-2-7-in-aws-lambda/) in March 2021 with plan to disable new function creation or updates Sep 30

**Comment by beingalitaheri** [comment]
Is there any solution for this yet?

**Comment by hyunsu110** [comment]
Do you have any update plans?
I want to use the cloud hub!!

**Comment by stealink** [comment]
Same issue 
Any update ?

**Comment by pattersonbl2** [comment]
We have submitted an updated template to AWS Marketplace it should be available in a few days.

**Reply by stealink** [reply]
Do you have an idea when it will be available ? We have a project to start and we are waiting to build our aws hubs entreprise

**Reply by stealink** [reply]
any update ? still in rollback

**Reply by pattersonbl2** [reply]
We are currently waiting for AWS to complete a review of the changes and add the template we uploaded.  This sometimes can be a lengthy process without a set eta.

**Reply by vjFred** [reply]
Any news on this?

**Reply by stealink** [reply]
I have try to launch a aws enterprise hubs yesterday and the same problem occurs

**Reply by vjFred** [reply]
@stealink, today on discord 
https://discord.com/channels/498741086295031808/819203046931693589/907252362157125694
someone gives me this link
https://github.com/mozilla/hubs-ops/blob/master/cloudformation/stack.yaml
But when I try to create the stack it gives me a Region error, maybe can you try to check if you have a better shot?

**Reply by stealink** [reply]
have you try with another region , because in the stack.yaml eu - west - 3 is missing .
I will try now, let me one hour

**Reply by vjFred** [reply]
Mine is us-east-1

**Reply by stealink** [reply]
same error for me 
08-11-2021 16:39:48 UTC+0100	hackinparis2021	ROLLBACK_COMPLETE	-
08-11-2021 16:39:40 UTC+0100	hackinparis2021	ROLLBACK_IN_PROGRESS	Template error: Unable to get mapping for Regions::eu-west-1::ImageId. Rollback requested by user.
08-11-2021 16:39:39 UTC+0100	hackinparis2021	CREATE_IN_PROGRESS	Template error: Unable to get mapping for Regions::eu-west-1::ImageId
08-11-2021 16:38:21 UTC+0100	hackinparis2021	CREATE_IN_PROGRESS	User Initiated

**Reply by vjFred** [reply]
Yes, and I have tried that on HC Personal and HC Enterprise, same error all the way...

**Reply by stealink** [reply]
I hope that it will be resolve soon , I have some event next week

**Reply by vjFred** [reply]
Mine is on Wednesday :(

**Reply by botissues** [reply]
sunday event here ;(  with 1 months of planning. I took it down because I couldnt SSH in and now I regret it because it was working fine

**Reply by botissues** [reply]
> Mine is on Wednesday :(

Its not looking good, sorry to hear about this

**Comment by pattersonbl2** [comment]
We are working to get a new CFT uploaded into the marketplace today.  Anyway having issues with this and cant wait.  You can download the 1.1.2 using the S3 bucket link amazon provides during launch setup and change these values to 0.0.1 
![unknown](https://user-images.githubusercontent.com/28871018/140782907-97869268-cb81-46f1-aaa4-ef68eabcd54a.png)

Also, make sure the Mapping Abbreviation are correct.
<img width="252" alt="Screen Shot 2021-11-08 at 11 46 24 AM" src="https://user-images.githubusercontent.com/28871018/140783259-c732a672-2169-403a-a9bb-5b8a4d09c8ea.png"

Sorry for any issues the process of updating CFT in the marketplace is a manual process on the AWS side which is creating a few errors and process break downs without little knowledge

**Reply by stealink** [reply]
follow the steps but still have error 


Transform AWS::Serverless-2016-10-31 failed with: Invalid Serverless Application Specification document. Number of errors found: 4. Resource with id [KeymasterApplication] is invalid. Application arn:aws:serverlessrepo:eu-west-1:753951066699:applications/keymaster-public-94ec4ef1-285d-4e63-b2eb-0b65ea93b942 does not have a version 0.0.1 Resource with id [NearsparkApplication] is invalid. Application arn:aws:serverlessrepo:eu-west-1:118096126957:applications/nearspark-94ec4ef1-285d-4e63-b2eb-0b65ea93b942 does not have a version 0.0.1 Resource with id [PhotomnemonicApplication] is invalid. Application arn:aws:serverlessrepo:eu-west-1:118096126957:applications/photomnemonic-94ec4ef1-285d-4e63-b2eb-0b65ea93b942 does not have a version 0.0.1 Resource with id [SpeelycaptorApplication] is invalid. Application arn:aws:serverlessrepo:eu-west-1:118096126957:applications/speelycaptor-94ec4ef1-285d-4e63-b2eb-0b65ea93b942 does not have a version 0.0.1. Rollback requested by user.

**Reply by stealink** [reply]
I use hubs entreprise

**Reply by vjFred** [reply]
same here

**Reply by vjFred** [reply]
@Pattersonbl2 
![image](https://user-images.githubusercontent.com/69004885/140789744-069ae350-a998-4428-a947-fef687b0446e.png)

**Reply by stealink** [reply]
any update ?  we have a event next week .

**Reply by pattersonbl2** [reply]
Currently, We are waiting for AWS to finish their publishing process for the new enterprise CFT. This is a manual process for AWS. They don't give us a timeline on when this will be complete so we are in the dark on an eta.  This error above is related to AWS not uploading the public lambdas in their marketplace repro yet .  reference the screenshot above and the hubs-ops repro to modify to directly enter the correct application id

**Reply by vjFred** [reply]
If any help is available please let us know, my event is tomorrow :(

**Reply by pattersonbl2** [reply]
You should be able to deploy a new Stack using the direction above. We update the cloud formation in this [repro](https://github.com/mozilla/hubs-ops/blob/master/cloudformation/stack.yaml) every time we make changes but the marketplace template has some injected values like AMI. I would download the template from the marketplace using the S3 URL and compare it to the repro to see what is missing.

**Reply by vjFred** [reply]
I have use [repro](https://github.com/mozilla/hubs-ops/blob/master/cloudformation/stack.yaml) but the deploy of a new Stack is not possible, using that stack.yaml file is giving me this error
![image](https://user-images.githubusercontent.com/69004885/140959321-8365df1a-934a-43cd-b347-a2773760d1a4.png)
and is not working as you mention

**Reply by pattersonbl2** [reply]
> I have use [repro](https://github.com/mozilla/hubs-ops/blob/master/cloudformation/stack.yaml) but the deploy of a new Stack is not possible, using that stack.yaml file is giving me this error
> 
> ![image](https://user-images.githubusercontent.com/69004885/140959321-8365df1a-934a-43cd-b347-a2773760d1a4.png)
> 
> and is not working as you mention 

Can you show me your stack in discord bp31.  I am sure it something off because we were able to re deploy stacks using same directions

**Reply by vjFred** [reply]
> > I have use [repro](https://github.com/mozilla/hubs-ops/blob/master/cloudformation/stack.yaml) but the deploy of a new Stack is not possible, using that stack.yaml file is giving me this error
> > ![image](https://user-images.githubusercontent.com/69004885/140959321-8365df1a-934a-43cd-b347-a2773760d1a4.png)
> > and is not working as you mention
> 
> Can you show me your stack in discord bp31. I am sure it something off because we were able to re deploy stacks using same directions

sorry, in discord? what can be off? i am not sure what are you asking me to do.., bp31? yes I can do, but what is that?

**Reply by stealink** [reply]
same issue here for a different region than @vjFred

**Reply by vjFred** [reply]
@stealink I get it... :) Brandon help me and I got it. Do you have discord? so I can share you my .yaml file. This one works for me and I am finally ONLINE region us-east1

**Reply by stealink** [reply]
Yes great news 

My discord is @marcito

**Reply by hungleon2112** [reply]
hi @vjFred , can you share me the .yaml file? My discord is: Leo2112.
Thank you!

**Reply by botissues** [reply]
> @stealink I get it... :) Brandon help me and I got it. Do you have discord? so I can share you my .yaml file. This one works for me and I am finally ONLINE region us-east1

Hello can you also share with me. discord:  Lilbbasedgod#1649

**Comment by pattersonbl2** [comment, marked as answer]
The enterprise stack has been updated. please feel free to redeploy stacks when ready
